components/core/src/clp_s/log_converter/log_converter.cpp

```diff
@@ -0,0 +1,143 @@
+#include <cstdint>
+#include <exception>
+#include <filesystem>
+#include <memory>
+#include <string_view>
+#include <system_error>
+#include <utility>
+
+#include <curl/curl.h>
+#include <spdlog/sinks/stdout_sinks.h>
+#include <spdlog/spdlog.h>
+
+#include "../../clp/NetworkReader.hpp"
+#include "../../clp/ReaderInterface.hpp"
+#include "../InputConfig.hpp"
+#include "CommandLineArguments.hpp"
+#include "LogConverter.hpp"
+
+using clp_s::log_converter::CommandLineArguments;
+using clp_s::log_converter::LogConverter;
+
+namespace {
+/**
+ * Checks for and logs CURL errors on a reader.
+ * @param path The path that the reader has opened.
+ * @param reader The open reader which may have experienced a CURL error.
+ * @return Whether a CURL error has occurred on the reader.
+ */
+auto check_and_log_curl_error(
+        clp_s::Path const& path,
+        std::shared_ptr<clp::ReaderInterface> const& reader
+) -> bool;
+
+/**
+ * Converts all files according to the command line arguments.
+ * @param command_line_arguments
+ * @return Whether conversion was successful.
+ */
+auto convert_files(CommandLineArguments const& command_line_arguments) -> bool;
+
+auto check_and_log_curl_error(
+        clp_s::Path const& path,
+        std::shared_ptr<clp::ReaderInterface> const& reader
+) -> bool {
+    if (auto network_reader = std::dynamic_pointer_cast<clp::NetworkReader>(reader);
+        nullptr != network_reader)
+    {
+        if (auto const rc = network_reader->get_curl_ret_code();
+            rc.has_value() && CURLcode::CURLE_OK != rc.value())
+        {
+            auto const curl_error_message = network_reader->get_curl_error_msg();
+            SPDLOG_ERROR(
+                    "Encountered curl error while converting {} - Code: {} - Message: {}",
+                    path.path,
+                    static_cast<int64_t>(rc.value()),
+                    curl_error_message.value_or("Unknown error.")
+            );
+            return true;
+        }
+    }
+    return false;
```

Comment:
```suggestion
    auto const network_reader = std::dynamic_pointer_cast<clp::NetworkReader>(reader);
    if (nullptr == network_reader) {
        return false;
    }
    if (auto const rc = network_reader->get_curl_ret_code();
        rc.has_value() && CURLcode::CURLE_OK != rc.value())
    {
        auto const curl_error_message = network_reader->get_curl_error_msg();
        SPDLOG_ERROR(
                "Encountered curl error while converting {} - Code: {} - Message: {}",
                path.path,
                static_cast<int64_t>(rc.value()),
                curl_error_message.value_or("Unknown error.")
        );
        return true;
    }
    return false;
```
Let's rewrite in this way.
btw, the converted `network_reader` can be `const`.